beacon-chain/verification/execution_payload_envelope.go

```diff
@@ -0,0 +1,205 @@
+package verification
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/signing"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v5/crypto/bls"
+	"github.com/prysmaticlabs/prysm/v5/time/slots"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	RequireBuilderValid Requirement = iota
+	RequirePayloadHashValid
+)
+
+// ExecutionPayloadEnvelopeGossipRequirements defines the list of requirements for gossip
+// execution payload envelopes.
+var ExecutionPayloadEnvelopeGossipRequirements = []Requirement{
+	RequireBlockRootSeen,
+	RequireBlockRootValid,
+	RequireBuilderValid,
+	RequirePayloadHashValid,
+	RequireSignatureValid,
+	RequireMessageNotSeen,
```

Comment:
This one is not validated. I think we need a new method like `VerifyMessageNotSeen` to validate this one.

Reply:
Yeah it's verified in the main routine here
```
	_, seen := s.payloadEnvelopeCache.Load(root)
	if seen {
		return pubsub.ValidationIgnore, nil
	}
```
This is outside the verification package, I'll remove the requirement